lib/test_track/attachment_settings.rb

```diff
@@ -0,0 +1,70 @@
+module TestTrack
+  class AttachmentSettings
+    include Singleton
+
+    class << self
+      delegate :storage_settings, :max_size, :attachments_enabled?, to: :instance
+    end
+
+    def attachments_enabled?
+      s3_enabled? || local_enabled?
+    end
+
+    def storage_settings
+      if s3_enabled?
+        s3_settings
+      else
+        local_settings
+      end
+    end
+
+    def max_size
+      ENV['ATTACHMENT_MAX_SIZE'] || 512.kilobytes
+    end
+
+    private
+
+    def s3_enabled?
+      storage_strategy == 's3' && s3_settings_valid?
```

Comment:
Should this do something squawkier if storage strategy is s3 and the s3 settings aren't valid? Maybe it starts up ok but drops an error in the log? If we did that, we'd probably want it (or the caller) to be memoized so it doesn't happen more than once per app initialization.